### PR TITLE
CI: raven github ci support

### PR DESCRIPTION
**Pull request by njucjc**

### CODE REVIEW

**Comment by DrmagicE** [comment]
How about we rename the directory `pkg/network-engine` to `pkg/networkengine`, and the package name to `networkengine` ?

**Comment by njucjc** [comment]
> How about we rename the directory `pkg/network-engine` to `pkg/networkengine`, and the package name to `networkengine` ?

done

**Comment by DrmagicE** [comment]
lgtm

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Check module vendoring
+        run: |
+          go mod tidy
+          go mod vendor
+          git diff --exit-code || (echo "please run 'go mod tidy && go mod vendor', and submit your changes"; exit 1)
```
Comment (DrmagicE) [inline .github/workflows/check.yml:31]: Do we need to submit the vendor directory?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Docker meta
```
Comment (DrmagicE) [inline .github/workflows/build.yml:39]: how about add unit test step before build？
```
      - name: Unit test
        run: go test ./...
```

File: .github/workflows/check.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+name: check
+
+on:
+  push: { }
```
[the remark this answers is not in the document]
Comment (njucjc) [reply .github/workflows/check.yml]: done.

**Pull request merged**